src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -115,33 +150,27 @@ def _execute(self, rseList):
                             number of RSEs cleaned
                             number of files deleted
         """
-        totalNumRses = 0
-        totalNumFiles = 0
-        numRsesCleaned = 0
-        numFilesDeleted = 0
 
-        # Call the workflow dispatcher:
-        for rse in rseList:
+        pline = self.plineUnmerged
+        self.resetCounters(plineName=pline.name)
+        self.plineCounters[pline.name]['totalNumRses'] = len(rseList)
```

Comment:
It is not that we have multiple pipelines, it is a patter that I was following. One has two major types of objects driving the service and each one has its counters - one is the RSE the other is the pipeline. The level of hierarchy in both cases is the same. The same also with the timers. It would be far from trivial to change this structure at this stage, because those are updated from several places, and the debugging would take quite a bit of time which we do not have.

Reply:
Yes, the RSE counter does make sense to be like that.
The pipeline one, is just an extra complication and making the code harder to read and maintain, given that there are no multiple pipelines and the service counters are always going to be in a flat structure.

I agree though not to make this change, it can potentially insert a bug if not carefully done and tested.